lib/matplotlib/scale.py

```diff
@@ -478,10 +478,110 @@ def get_transform(self):
         return self._transform
 
 
+def _mask_non_logit(a):
+    """
+    Return a Numpy masked array where all values outside ]0, 1[ are
+    masked. If all values are inside ]0, 1[, the original array is
+    returned.
+    """
+    mask = (a <= 0.0) | (a >= 1.0)
+    if mask.any():
+        return ma.MaskedArray(a, mask=mask)
```

Comment:
I would just do `a[mask] = np.nan` .  I think we are trying to move away from using masked arrays internally due to the issues with them.  Again, make sure a copy gets made before mutating a user object.